sql/src/main/java/io/crate/operation/aggregation/impl/PercentileAggregation.java

```diff
@@ -110,20 +104,18 @@ private TDigestState initState(Object argValue) {
 
     @Override
     public TDigestState reduce(RamAccountingContext ramAccountingContext, TDigestState state1, TDigestState state2) {
-        if (state1 == null) {
-            return state2;
-        }
-        if (state2 != null) {
+        if (state1.isEmpty()) {
             state1.add(state2);
```

Comment:
why not use state2? i think the logic should stay as it was, just use isEmpty instead of then null check?